midas-contracts/contracts/RedemptionVault.sol

```diff
@@ -325,14 +325,18 @@ contract RedemptionVault is ManageableVault, IRedemptionVault {
 
         mToken.burn(address(this), request.amountMToken);
 
-        if (request.tokenOut != MANUAL_FULLFILMENT_TOKEN) {
-            uint256 tokenDecimals = _tokenDecimals(request.tokenOut);
+        bool isFiat = request.tokenOut == MANUAL_FULLFILMENT_TOKEN;
 
-            uint256 amountTokenOutWithoutFee = _truncate(
-                (request.amountMToken * newMTokenRate) / request.tokenOutRate,
-                tokenDecimals
-            );
+        uint256 tokenDecimals = isFiat ? 18 : _tokenDecimals(request.tokenOut);
 
+        uint256 amountTokenOutWithoutFee = _truncate(
+            (request.amountMToken * newMTokenRate) / request.tokenOutRate,
+            tokenDecimals
+        );
+
+        _requireAndUpdateAllowance(request.tokenOut, amountTokenOutWithoutFee);
```

Comment:
Just to double check, the latest design is that fiat tokens are also capped with an allowance?

Reply:
yep